lib/common.js

```diff
@@ -82,3 +82,26 @@ exports.assert.shouldThrowGen = function* (func, errorObject, message) {
   if (message === undefined) message = '';
   chai.assert(false, `${message}. Should have thrown error: ${errorObject.toString()} instead got ${JSON.stringify(result)}`);
 }
+
+/* execute a generator function that retry the input func until,
+ the func is executed with some return value or timeout with expected error */
+exports.assert.retry = function* (func, timeoutSeconds, expectedError) {
```

Comment:
Retry a generator function (func) for a given timeout (timeoutSeconds). Allow an optional error (expectedError).

Reply:
Done

Reply:
Done